core/src/main/scala/org/apache/spark/shuffle/BlockStoreShuffleReader.scala

```diff
@@ -35,11 +35,35 @@ private[spark] class BlockStoreShuffleReader[K, C](
     readMetrics: ShuffleReadMetricsReporter,
     serializerManager: SerializerManager = SparkEnv.get.serializerManager,
     blockManager: BlockManager = SparkEnv.get.blockManager,
-    mapOutputTracker: MapOutputTracker = SparkEnv.get.mapOutputTracker)
+    mapOutputTracker: MapOutputTracker = SparkEnv.get.mapOutputTracker,
+    fetchMultiPartitions: Boolean)
```

Comment:
to reduce diff, how about we give it a default value `true`? Existing tests won't enable batch fetch as they don't set the local property.

Reply:
We can even remove this parameter. The caller side can drop the special local property if it doesn't need to fetch multiple partitions.

Reply:
hmmm, it's not good to handle the special local property in too many places. How about we read the local property in the caller side, check the # of partitions need to fetch, and pass a single "shouldBatchFetch" flag to `BlockStoreShuffleReader`?

Reply:
Yeah, pick the last approach, that's clearer, also move the property's key into SortShuffleManager.